src/indexer/merger.rs

```diff
@@ -284,25 +287,27 @@ impl IndexMerger {
     }
 
 
-    fn write_multi_fast_field_idx(&self,
-                                  field: Field,
-                                  fast_field_serializer: &mut FastFieldSerializer) -> Result<()> {
+    fn write_fast_field_idx(&self,
+                            field: Field,
+                            fast_field_serializer: &mut FastFieldSerializer) -> Result<()> {
         let mut total_num_vals = 0u64;
 
         // In the first pass, we compute the total number of vals.
         //
         // This is required by the bitpacker, as it needs to know
         // what should be the bit length use for bitpacking.
         for reader in &self.readers {
-            let multi_ff_reader = reader.multi_fast_field_reader::<u64>(field)?;
+            let idx_reader = reader.fast_field_reader_with_idx::<u64>(field, 0)?;
             if let Some(delete_bitset) = reader.delete_bitset() {
                 for doc in 0u32..reader.max_doc() {
                     if !delete_bitset.is_deleted(doc) {
-                        total_num_vals += multi_ff_reader.num_vals(doc) as u64;
+                        let start = idx_reader.get(doc);
+                        let end = idx_reader.get(doc + 1);
```

Comment:
why did you remove `.num_vals` ?

Reply:
Because it's no longer used. I could move it to FastFieldReader<u64> (which is all this method uses now), but it's no longer used by anything on the multi reader.

Reply:
ok I see